packages/rocketchat-api/server/routes.coffee

```diff
@@ -185,5 +196,195 @@ RocketChat.API.v1.addRoute 'groups.create', authRequired: true,
 			return RocketChat.API.v1.failure e.name + ': ' + e.message
 
 		return RocketChat.API.v1.success
-			group: RocketChat.models.Rooms.findOne({_id: id.rid})
+			group: RocketChat.models.Rooms.findOne({_id: id.rid})		
+
+
+#list All Private Groups		
+RocketChat.API.v1.addRoute 'privateRooms.list', authRequired: true,
+	get: ->
+		try
+			this.response.setTimeout (1000)
+			rooms = RocketChat.models.Rooms.findByType('p', { sort: { msgs:-1 } }).fetch()
```

Comment:
We should verify some permissions, maybe we should create new permissions for these methods like `api-groups-list` that only admins have this permission